IDEOorg/steps-bot
# #162669788 As a developer, I should be able to investigate details of an error thrown in the bot code by looking in a date-stamped file on the bot server

### What does this PR do?
* Write error to file with datestamp

### Description of tasks to be completed
* Add log4js package to handle writing logs to files
* Replace `console.log` statements with log4js methods

### How should this be manually tested?
* Clone this repo
* Checkout to a branch `ft-write-error-to-file-with-datestamp-162669788`.
* Create an error by removing the `OAUTH_ACCESS_TOKEN` from the `.env` file.
* Run `npm start` and `npm run start:live:dev`
* [Notice that a `log` directory was created
* Setup Twilio webhook.
* Log in as a coach and create a client if not already present.
* Add a task to the client's work plan.
* Visit [twilio-gui](http://twilio-gui.herokuapp.com) and get a phone number.
* login to `twilio-gui`
* Initiate the conversation with the bot by sending `start`.
* Check the console on your local terminal for the error feedback.
* Check that the console error is written to the `error.log` file in the `log` directory

### What are the relevant pivotal tracker stories?
[#162669788](https://www.pivotaltracker.com/story/show/162669788)

### Any Background Context
_"The bot server does not use any error logging system currently. We want to record all important details related to errors that occur so that developers can use the data to troubleshoot bugs. The suggested approach is to create error files."_ - @matthewepler 

### Screenshots?
![write-error-log-to-file1](https://user-images.githubusercontent.com/31534129/52655039-8bdda480-2ef3-11e9-814b-e057f0a8a52d.gif)

![write-error-log-to-file1](https://user-images.githubusercontent.com/31534129/52655254-0dcdcd80-2ef4-11e9-9fa5-7b1c38a93f0c.gif)




### Questions
N/A
